src/Security/Member.php

```diff
@@ -1491,8 +1491,11 @@ private function scaffoldFormFieldForManyRelation(string $relationName, ?string
         $field = SearchableMultiDropdownField::create($relationName, $fieldTitle, $list);
         // Use the same lazyload threshold has_one relations use
         $threshold = DBForeignKey::config()->get('dropdown_field_threshold');
-        $overThreshold = $list->count() > $threshold;
-        $field->setIsLazyLoaded($overThreshold)->setLazyLoadLimit($threshold);
+        $overThreshold = $threshold === 0 || $list->setUseCache(true)->count() > $threshold;
+        $field->setIsLazyLoaded($overThreshold);
+        if ($threshold > 0) {
+            $field->setLazyLoadLimit($threshold);
+        }
```

Comment:
The checks against `$threshold` being 0 mirror what was done in https://github.com/silverstripe/silverstripe-framework/pull/11646 for `DataObject::scaffoldFormFieldForHasOne()`